scripts/optionsManager.js

```diff
@@ -215,7 +215,7 @@ class OptionsManager {
 	// FUNCTIONS
 
 	async sleep(time) {
-		return await new Promise((resolve) => {
```

Comment:
This is just a technicality, but the `await` isn't needed here.
Using `await` here means that the function sleep takes `time` ms to return `undefined`. You then `await undefined` in every call of the function (which just resolves to `undefined`). Your usage of the function thus results in one redundant `await` and looks like this one here was a mistake.

Reply:
Yep, I wrote this some time ago when I didn't really understood Promises, I guess :)